mods/post_data.js

```diff
@@ -0,0 +1,124 @@
+"use strict";
+
+/**
+ * Key-value bucket handler
+ */
+
+// TODO: move to separate spec package
+var yaml = require('js-yaml');
+var fs = require('fs');
+var spec = yaml.safeLoad(fs.readFileSync(__dirname + '/post_data.yaml'));
+var crypto = require('crypto');
+var stringify = require('json-stable-stringify');
+var URI = require('swagger-router').URI;
+
+function PostDataBucket(options) {
+    this.log = options.log || function() {};
+}
+
+PostDataBucket.prototype.getBucketInfo = function(restbase, req) {
+    var rp = req.params;
+    return restbase.get({
+        uri: new URI([rp.domain, 'sys', 'key_value', rp.bucket]),
+    });
+};
+
+PostDataBucket.prototype.createBucket = function(restbase, req) {
+    var rp = req.params;
+    return restbase.put({
+        uri: new URI([rp.domain, 'sys', 'key_value', rp.bucket]),
+        body: {
+            keyType: 'string',
+            valueType: 'json'
+        }
+    });
+};
+
+PostDataBucket.prototype.listBucket = function(restbase, req) {
+    var rp = req.params;
+    return restbase.get({
+        uri: new URI([rp.domain, 'sys', 'key_value', rp.bucket, ''])
+    });
+};
+
+PostDataBucket.prototype.getRevision = function(restbase, req) {
+    var rp = req.params;
+    var path = [rp.domain, 'sys', 'key_value', rp.bucket, rp.key];
+    if (rp.tid) {
+        path.push(rp.tid);
+    }
+    return restbase.get({
+        uri: new URI(path),
+        headers: {
+            'cache-control': req.headers && req.headers['cache-control']
+        }
+    });
+};
+
+function prepareStoredReq(req) {
+    var storedReq = {};
+    storedReq.uri = req.body && req.body.uri && req.body.uri.toString();
```

Comment:
@d00rman I could imagine a useful empty `req.body.body`. It could represent something like an empty graph, or an empty table (if we had some table-rendering service). So, although it's not used now, it's possible that empty body would be a legal request for some service.